From bc641236bd9472d206e6900332542c4b6d36e2a3 Mon Sep 17 00:00:00 2001
From: Eric Biggers <ebiggers@google.com>
Date: Sat, 26 Mar 2022 01:08:07 +0000
Subject: [PATCH] [RESTRICT AUTOMERGE] Log to EventLog on prepareUserStorage
 failure

Bug: 224585613
Change-Id: Id6dfb4f4c48d5cf4e71f54bdb6d0d6eea527caf5
(cherry picked from commit fbb632ea95ac5b6d9efa89e09d0988a9df4f19e4)
Merged-In: Id6dfb4f4c48d5cf4e71f54bdb6d0d6eea527caf5
(cherry picked from commit 0762961674f1454b7c7012a0ab53c427570e836c)
Merged-In: Id6dfb4f4c48d5cf4e71f54bdb6d0d6eea527caf5
---
 .../core/java/com/android/server/StorageManagerService.java     | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index 64aa443bdda..eb65cac43c6 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -131,6 +131,7 @@ import android.util.ArrayMap;
 import android.util.ArraySet;
 import android.util.AtomicFile;
 import android.util.DataUnit;
+import android.util.EventLog;
 import android.util.FeatureFlagUtils;
 import android.util.Log;
 import android.util.Pair;
@@ -3381,6 +3382,7 @@ class StorageManagerService extends IStorageManager.Stub
                 }
             }
         } catch (Exception e) {
+            EventLog.writeEvent(0x534e4554, "224585613", -1, "");
             Slog.wtf(TAG, e);
             // Make sure to re-throw this exception; we must not ignore failure
             // to prepare the user storage as it could indicate that encryption
-- 
2.36.1.476.g0c4daa206d-goog

